src/loader.js

```diff
@@ -15,40 +16,34 @@ module.exports = function(source) {
 
   const callback = webpack.async();
 
-  global.__DEBUG__ = (
-    process.env.DEBUG === 'sass-resources-loader' || process.env.DEBUG === '*'
-  );
+  global.__DEBUG__ = process.env.DEBUG === 'sass-resources-loader' || process.env.DEBUG === '*';
 
   logger.debug(`Hey, we're in DEBUG mode! Yabba dabba doo!`);
 
-  const resourcesLocation = parseResources(webpack.options.sassResources);
+  // TODO: Remove `webpack.options.sassResources` support after first stable webpack@2 release
+  const resourcesFromConfig =
+    webpack.version !== 2
+    ? webpack.options.sassResources
+    : loaderUtils.parseQuery(this.query).resources
+  ;
+  const resourcesLocations = parseResources(resourcesFromConfig);
   const moduleContext = webpack.context;
   const webpackConfigContext = webpack.options.context;
 
   logger.debug('Module context:', moduleContext);
   logger.debug('Webpack config context:', webpackConfigContext);
+  logger.debug('Resources:', resourcesLocations);
 
-  if (!resourcesLocation) {
-    const error = new Error(`
-      Could not find sassResources property.
-      Make sure it's defined in your webpack config.
-    `);
-
-    return callback(error);
-  }
-
-  logger.debug('sassResources:', resourcesLocation);
-
-  if (!resourcesLocation.length) {
+  if (!resourcesLocations.length) {
     const error = new Error(`
-      Looks like sassResources property has wrong type.
-      Make sure it's String or Array of Strings.
+      Something wrong with provided resources.
+      Make sure 'options.resources' is String or Array of Strings.
     `);
 
     return callback(error);
   }
 
-  const files = resourcesLocation.map(resource => {
+  const files = resourcesLocations.map(resource => {
     const file = path.resolve(webpackConfigContext, resource);
```

Comment:
@avgalon Thanks for the feedback! Alex (implementor of this PR is on vacation until Monday). I'm not sure if he'll see before that time, but I'll make sure that he does when gets back on Monday as we definitely need to get this addressed.